WalletWasabi/WabiSabi/Backend/Rounds/Arena.cs

```diff
@@ -91,10 +92,33 @@ protected override Task ActionAsync(CancellationToken cancel)
 				throw new WabiSabiProtocolException(WabiSabiProtocolErrorCode.TooMuchWeight);
 			}
 
-			return round.RegisterAlice(
+			var resp = round.RegisterAlice(
 				alice,
 				zeroAmountCredentialRequests,
 				zeroWeightCredentialRequests);
+
+			lock (Lock)
```

Comment:
This locking doesn't feel right although I know it's correct, it kinda puts locks all over the place.